hw10/hw10_t01.py

```diff
@@ -0,0 +1,225 @@
+                    "a", span="", href=re.compile("stock$"), attrs={"title": True}
+                )
+
+                for company in stocks:
+                    growth = float(
+                        list(list(company.parent.next_siblings)[-4].children)[
+                            -2
+                        ].string.rstrip("%")
+                    )
+                    company_info = {
+                        "name": company["title"],
+                        "code": None,
+                        "price": None,
+                        "P/E": None,
+                        "growth": growth,
+                        "potential profit": None,
+                    }
+
+                    url = base_company_url + company["href"]
+
+                    (
+                        company_info["code"],
+                        company_info["price"],
+                        company_info["P/E"],
+                        company_info["potential profit"],
+                    ) = await get_company_info(url, session, usd)
+
+                    evaluate_top10(company_info)
+    write_into_files()
+
+
+async def get_usd_currency_from_cbr(session: aiohttp.ClientSession) -> float:
+    today_date = dt.datetime.strftime(dt.datetime.today(), "%d/%m/%Y")
+    yesterday_date = dt.datetime.strftime(
+        dt.datetime.today() - dt.timedelta(days=1), "%d/%m/%Y"
+    )
+    usd_params = {
+        "date_req1": yesterday_date,
+        "date_req2": today_date,
+        "VAL_NM_RQ": "R01235",
+    }
+    async with session.get(
+        "http://www.cbr.ru/scripts/XML_dynamic.asp", params=usd_params
+    ) as resp:
+        html = await resp.text()
+        soup = BeautifulSoup(html, features="html.parser")
+        return float(soup.find_all("value")[-1].string.replace(",", "."))
+
+
+async def get_company_info(
+    url: str, session: aiohttp.ClientSession, usd: float
+) -> tuple:
+    async with session.get(url) as resp:
+        html = await resp.text()
+        soup = BeautifulSoup(html, features="html.parser")
+
+        current_value = float(
+            soup.find(
+                "span", attrs={"class": "price-section__current-value"}
+            ).string.replace(",", "")
+        )
+
+        code = list(
+            soup.find("span", attrs={"class": "price-section__category"}).strings
+        )[-2].strip(", ")
+
+        try:
+            p_e = float(
+                list(
+                    soup.find(
+                        "div", attrs={"class": "snapshot__header"}, string="P/E Ratio"
+                    ).parent.stripped_strings
+                )[0].replace(",", "")
+            )
+        except AttributeError:
+            p_e = None
+
+        try:
+            year_low = float(
+                list(
+                    soup.find(
+                        "div", attrs={"class": "snapshot__header"}, string="52 Week Low"
+                    ).parent.stripped_strings
+                )[0].replace(",", "")
+            )
```

Comment:
This code looks familiar.  I have already seen it in this file. Could we do something about it?